File: deepseek/3.Desarrollo Web Endpoint RESTful/golang/endpoint.go
package main

import "fmt"

type Singleton struct{}

var instance *Singleton

func GetInstance() *Singleton {
    if instance == nil {
        instance = &Singleton{}
    }
    return instance
}

func main() {
    s1 := GetInstance()
    s2 := GetInstance()
    fmt.Println("¿s1 y s2 son la misma instancia?", s1 == s2)
}

File: deepseek/3.Desarrollo Web Endpoint RESTful/c++/endpoint.cpp
#include <iostream>
using namespace std;

class Singleton {
private:
    static Singleton* instance;
    Singleton() {}

public:
    static Singleton* getInstance() {
        if (instance == nullptr) {
            instance = new Singleton();
        }
        return instance;
    }
};

Singleton* Singleton::instance = nullptr;

int main() {
    Singleton* s1 = Singleton::getInstance();
    Singleton* s2 = Singleton::getInstance();
    cout << "¿s1 y s2 son la misma instancia? " << (s1 == s2) << endl;
    return 0;
}